x/gamm/keeper/grpc_query.go

```diff
@@ -422,7 +422,10 @@ func (q Querier) EstimateSwapExactAmountIn(ctx context.Context, req *types.Query
 
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 
-	tokenOutAmount, err := q.Keeper.MultihopEstimateOutGivenExactAmountIn(sdkCtx, req.Routes, tokenIn)
+	// TODO: remove this redundancy after making routes be shared between x/gamm and x/swaprouter.
+	swaprouterRoutes := types.ConvertAmountInRoutes(req.Routes)
```

Comment:
Note: this implies proto-generating routes only in swaprouter and importing them in x/gamm.

According to earlier discussions, this should not cause breakage and will be done in a future PR

Reply:
SG if were confident it wont break stuff. Is there an issue or something written to track our testing strategy? (Not PR merge blocking)

Reply:
According to our earlier discussions, this should not be state-breaking. 

The only concern was breaking telescope. I'm planning to make a follow-up PR and tag Dan Lynch to help with looking into this.

I can also add a test to serialize both old and new routes to make sure that they are valid

Reply:
https://github.com/osmosis-labs/osmosis/issues/3808